students/lauraannf/lessons/lesson06/assignment/good_perf.py

```diff
@@ -0,0 +1,59 @@
+"""
+poorly performing, poorly written module
+
+"""
+
+import datetime
+import csv
+import time
+
+
+def analyze(filename):
+    """open and analyze csv file"""
+    start = datetime.datetime.now()
+    start_open1 = time.clock()
+    with open(filename) as csvfile:
+        reader = csv.reader(csvfile, delimiter=',', quotechar='"')
+        found = 0
+        year_count = {
+            "2013": 0,
+            "2014": 0,
+            "2015": 0,
+            "2016": 0,
+            "2017": 0,
+            "2018": 0
+        }
+        for row in reader:
+            if row[5] > '00/00/2012':
```

Comment:
A simpler way to count occurrences per year: Capture the year and see if it's present in your _year_count_ dictionary:

```
            if row[5][6:] in year_count:
                year_count[row[5][6:]] += 1
```

Reply:
This also translates into ~30% reduction from your already optimized run time.